### PR TITLE
Adding support for opentracing 0.32

### PR DESCRIPTION
**Pull request by dougEfresh**
WIP  for opentracing 0.32
See discussions at:
https://github.com/opentracing/opentracing-java/issues/314

See also jaeger issue #566

### CODE REVIEW

File: jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TextMapCodecTest.java [hunk 1]
```diff
@@ -132,31 +134,56 @@ public void testContextAsStringFormatsPositiveFields() {
...
     headers.put("jaeger-baggage", "k1=v1, k2 = v2, k3=v3=d3");
     TextMapCodec codec = new TextMapCodec(false);
-    JaegerSpanContext context = codec.extract(new TextMapExtractAdapter(headers));
+    JaegerSpanContext context = codec.extract(textMap);
     assertEquals("v1", context.getBaggageItem("k1"));
     assertEquals("v2", context.getBaggageItem("k2"));
     assertEquals(null, context.getBaggageItem("k3"));
   }
+
+  private static class TestTextMap implements TextMap {
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TextMapCodecTest.java]: same class 3rd time? we need to DRY it

File: jaeger-core/src/test/java/io/jaegertracing/internal/AdhocHeadersTest.java [hunk 2]
```diff
@@ -83,11 +84,34 @@ public void testJoinTraceWithAdhocBaggage() {
   private void traceWithAdhocBaggage(Map<String, String> headers) {
     headers.put("jaeger-baggage", "k1=v1, k2 = v2");
 
-    JaegerSpanContext parent = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(headers));
+    JaegerSpanContext parent = tracer.extract(Format.Builtin.HTTP_HEADERS, new TestTextMap(headers));
```
[the remark this answers is not in the document]
Comment (dougEfresh) [reply jaeger-core/src/test/java/io/jaegertracing/internal/AdhocHeadersTest.java]: Open tracing changed the signature of [TextMap](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/propagation/TextMap.java)

File: jaeger-core/src/test/java/io/jaegertracing/ConfigurationTest.java [hunk 3]
```diff
@@ -479,14 +479,15 @@ public void testCodecWithPropagationB3() {
   @SuppressWarnings("unchecked")
   private <C> void assertInjectExtract(JaegerTracer tracer, Format<C> format, JaegerSpanContext contextToInject,
                                        boolean injectMapIsEmpty) {
-    HashMap<String, String> injectMap = new HashMap<>();
-    tracer.inject(contextToInject, format, (C) new TextMapInjectAdapter(injectMap));
+    Map<String, String> injectMap = new HashMap<>();
+    TextMap textMap = new TestTextMap(injectMap);
+    tracer.inject(contextToInject, format, (C) textMap);
```
[the remark this answers is not in the document]
Comment (dougEfresh) [reply jaeger-core/src/test/java/io/jaegertracing/ConfigurationTest.java]: I didn't realize they created a new [TextMapAdaptor](https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/propagation/TextMapAdapter.java).
 I will use that.

File: jaeger-core/src/test/java/io/jaegertracing/internal/AdhocHeadersTest.java [hunk 4]
```diff
@@ -83,11 +84,34 @@ public void testJoinTraceWithAdhocBaggage() {
...
     JaegerSpan span = tracer.buildSpan("test").asChildOf(parent).start();
 
     assertTrue(span.context().isSampled());
     assertEquals("must have baggage", "v1", span.getBaggageItem("k1"));
     assertEquals("must have baggage", "v2", span.getBaggageItem("k2"));
   }
+
+  private static class TestTextMap implements TextMap {
```
[the remark this answers is not in the document]
Comment (dougEfresh) [reply jaeger-core/src/test/java/io/jaegertracing/internal/AdhocHeadersTest.java]: Will remove

File: jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java [hunk 5]
```diff
@@ -74,6 +75,10 @@ public long getStart() {
     return startTimeMicroseconds;
   }
 
+  public boolean isFinished() {
+    return finished;
+  }
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java]: why is this needed?

File: jaeger-zipkin/src/test/java/io/jaegertracing/zipkin/internal/ThriftSpanConverterTest.java [hunk 6]
```diff
@@ -318,4 +316,31 @@ public void testConvertSpanWith128BitTraceId() {
     assertEquals(span.context().getTraceIdLow(), zipkinSpan.getTrace_id());
     assertEquals(span.context().getTraceIdHigh(), zipkinSpan.getTrace_id_high());
   }
+
+  static class TestTextMap implements TextMap {
```
Comment (yurishkuro) [inline jaeger-zipkin/src/test/java/io/jaegertracing/zipkin/internal/ThriftSpanConverterTest.java]: can we use TextMapAdaptor here too?

File: jaeger-zipkin/src/test/java/io/jaegertracing/zipkin/internal/V2SpanConverterTest.java [hunk 7]
```diff
@@ -311,4 +311,27 @@ public void testConvertSpanWith128BitTraceId() {
     assertNotEquals(0, span.context().getTraceIdHigh());
     assertEquals(span.context().getTraceId(), zipkinSpan.traceId());
   }
+
+  static class TestTextMap implements TextMap {
```
Comment (yurishkuro) [inline jaeger-zipkin/src/test/java/io/jaegertracing/zipkin/internal/V2SpanConverterTest.java]: can we use TextMapAdaptor here too?

File: jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java (hunk 5 above)
Comment (dougEfresh) [reply jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java]: We use a custom async/future framework at my work. We pass spans to the framework which will activate the span on ThreadLocal scope when futures will be executed. 
Sometimes the futures will have stale or old spans which are not cleaned up properly (mostly due to misuse of the framework, or cached futures...etc). 
I have a custom ScopeManager which detects a span isFinished() but yet will be activated on a ThreadLocal scope. I use a counter metric to record when this occurs. 
Without this method available, I would (have) never know this even occurred.

Comment (yurishkuro) [reply jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java]: Could you please create a separate issue to discuss this, and not include it in this PR?

Comment (dougEfresh) [reply jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java]: sure thing
